core/src/utils/test/playwright/playwright-page.ts

```diff
@@ -31,24 +31,42 @@ type CustomFixtures = {
   page: E2EPage;
 };
 
-export const test = base.extend<CustomFixtures>({
-  page: async ({ page }: CustomTestArgs, use: (r: E2EPage) => Promise<void>, testInfo: TestInfo) => {
-    const originalGoto = page.goto.bind(page);
-    const originalLocator = page.locator.bind(page);
-
-    // Overridden Playwright methods
-    page.goto = (url: string) => goToPage(page, url, testInfo, originalGoto);
-    page.setContent = (html: string) => setContent(page, html);
-    page.locator = (selector: string, options?: LocatorOptions) => locator(page, originalLocator, selector, options);
-    // Custom Ionic methods
+/**
+ * Extends the base `page` test figure within Playwright.
+ * @param page The page to extend.
+ * @param testInfo The test info.
+ * @returns The modified playwright page with extended functionality.
+ */
+export async function extendPageFixture(page: E2EPage, testInfo?: TestInfo) {
+  const originalGoto = page.goto.bind(page);
+  const originalLocator = page.locator.bind(page);
+
+  // Overridden Playwright methods
+
+  if (testInfo) {
+    page.goto = (url: string, options) => goToPage(page, url, options, testInfo, originalGoto);
+  }
+  page.setContent = (html: string) => setContent(page, html);
+  page.locator = (selector: string, options?: LocatorOptions) => locator(page, originalLocator, selector, options);
+
+  // Custom Ionic methods
+
+  if (testInfo) {
```

Comment:
This is confusing. `testInfo` should always be available when running E2E tests in Ionic Core. Additionally, the `E2EPage` interface says that these methods are always available, but that would no longer be the case with this change.

Reply:
Agreed, these checks can be removed now. Originally I was trying to extend the `browser` fixture when the `newPage()` from the browser context is called, which didn't readily have access to test info. But, with exporting the function, the calling test suite should always have a page and test info context.